### ISSUE TITLE
Request: Ability to mass delete entries based on filters

### ISSUE THREAD

**Issue by darylknight** [verbb/wishlist#143]
### What are you trying to do?

I've just checked in on a site with Wishlist that we created a year ago and noticed there are 55,000 entries. Most of the wishlists have zero items in them, so it would be nice to be able to clear them out to stop the database from getting larger over time.

Currently, there is no way to filter the element index view based on things like "Items" (presumably because it's a custom field).

Any chance of some kind of functionality to be able to mass clear out lists?

### What's your proposed solution?

Either a filter for things like "Items" so I can use native functionality to select all > delete, or inbuilt functionality to mass-delete wishlists.

### Additional context

![image](https://github.com/user-attachments/assets/7a4a4e8b-db9b-4fc4-a244-21c6a85893c0)

**Comment by engram-design** [comment]
There's a few [options](https://verbb.io/craft-plugins/wishlist/docs/get-started/configuration) already to help with this:

- `purgeInactiveLists` - Whether to purge inactive lists after a certain duration.
- `purgeInactiveListsDuration` - If purging inactive lists is enabled, after this duration they will be purged.
- `purgeInactiveGuestListsDuration` - If purging inactive lists is enabled, after this duration only guest lists will be purged.
- `purgeEmptyListsOnly` - Whether to purge user lists only if they have no items.
- `purgeEmptyGuestListsOnly` - Whether to purge guest lists only if they have no items.

All of these will be configured to run automatically, to help your site's lists/items from getting out of hand.

There's also a console command `./craft wishlist/lists/purge-inactive-lists`.

Otherwise I can definitely add some filtering and even just ordering by "Items" (which is just a count).

**Comment by darylknight** [comment]
That's great, thanks Josh. One question on that - the docs say "The below shows the defaults already used by Wishlist, so you don't need to add these options unless you want to modify the values."

The defaults are supposed to clear out empty guest wishlists after 1 day. We don't have registstered users on the site so they're all guest wishlists, and we have thousands of empty lists on the site from a year ago. Any idea why?

I also tried adding these settings to a wishlist.php file and they're not getting cleared out.

Also - the control panel is missing the "purgeInactiveListsDuration" option

**Comment by engram-design** [comment]
You didn't mention what version you're currently on, but [2.0.12](https://github.com/verbb/wishlist/releases/tag/2.0.12) (the latest for Craft 4) should have these set by default. And if you have a `config/wishlist.php` file, and those values aren't (or are) set, then that's going to take precedence.

It should be stated that it all works through Craft's garbage collection as well, so it's not always guaranteed to run, but it should often-ish. Running a `./craft gc` will kick it into gear.

And yep, these are just config-based settings for the moment.

**Comment by darylknight** [comment]
I updated it on Monday to 2.0.12 - it would have been 6+ months since the previous update before that.

So, I just ran `./craft gc` manually and it went from 56,800 to 56,200 - but there are still thousands of empty lists in there from 2023.

Contents of `wishlist.php`:

```
<?php

return [
        '*' => [
                'pluginName' => 'Wishlist',
                'showListInfoTab' => false,

                // Lists
                'allowDuplicates' => false,
                'manageDisabledLists' => true,
                'mergeLastListOnLogin' => false,
                'purgeInactiveLists' => true,
                'purgeInactiveListsDuration' => 'P1M', // 3 months
                'purgeInactiveGuestListsDuration' => 'P7D', // 7 days
                'purgeEmptyListsOnly' => true,
                'purgeEmptyGuestListsOnly' => true,
                'cookieExpiry' => 0, // Session-based

                // PDF
                'pdfFilenameFormat' => 'Wishlist-{id}',
                'pdfPath' => '_pdf/wishlist',
                'pdfAllowRemoteImages' => true,
                'pdfPaperSize' => 'letter',
                'pdfPaperOrientation' => 'portrait',

                // Email
                'templateEmail' => null,
                'attachPdfToEmail' => false,
        ],
];
```

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, run `composer require verbb/wishlist:"dev-craft-4 as 2.0.12"`. Do also note that if you want a purely time-based purge, you'll want to disable `purgeEmptyListsOnly ` and `purgeEmptyGuestListsOnly `.

**Comment by engram-design** [comment]
Fixed in [2.0.13](https://github.com/verbb/wishlist/releases/tag/2.0.13)
